features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -10,54 +10,56 @@ import android.annotation.SuppressLint
 import android.os.Build
 import android.view.View
 import android.view.inspector.WindowInspector
+import com.datadog.android.api.InternalLogger
 import java.lang.reflect.Field
 
 @SuppressLint("PrivateApi")
 internal object WindowInspector {
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    internal const val FAILED_TO_RETRIEVE_DECOR_VIEWS_ERROR_MESSAGE =
+        "SR WindowInspector failed to retrieve the decor views"
+
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_CLASS by lazy {
-        try {
-            return@lazy Class.forName("android.view.WindowManagerGlobal")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy Class.forName("android.view.WindowManagerGlobal")
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_INSTANCE by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val VIEWS_FIELD by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
     }
 
-    fun getGlobalWindowViews(): List<View> {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            WindowInspector.getGlobalWindowViews()
-        } else {
-            getGlobalWindowViewsLegacy(GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+    @SuppressWarnings("TooGenericExceptionCaught")
+    fun getGlobalWindowViews(internalLogger: InternalLogger): List<View> {
+        return try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                WindowInspector.getGlobalWindowViews()
+            } else {
+                getGlobalWindowViewsLegacy(internalLogger, GLOBAL_WM_INSTANCE, VIEWS_FIELD)
```

Comment:
We already catch exceptions in that method, shouldn't we instead have two methods (e.g.: `getGlobalWindowViewsLegacy` and `getGlobalWindowViewsAPI29`) throwing exceptions, and having a global catch here? Or having each method catch their own exceptions?

Reply:
yes I forgot to remove the one from `getGlobalWindowLegacy` . My idea was the same as your but actually there is no point of `getGlobalWindowViewsAPI29` as that is only calling the system method. We can catch everything here but indeed need to remove the catch from below.